src/main/java/frc/robot/VisionProcessor.java

```diff
@@ -132,19 +145,24 @@ public void process(Mat sourceFrame) {
         ArrayList<RotatedRect> initial = minimumBoundingRectangle(parent.filterContoursOutput());
         ArrayList<RotatedRect> findContours = minimumBoundingRectangle(parent.findContoursOutput());
         ArrayList<RotatedRect> ok = new GibberishRectangleFilter().filter(initial);
-        ArrayList<RotatedRect> selected = new DumbAndAloneRectangleFilter().filter(ok);
-
+        index = new CenterRectangleIndexFinder().getIndex(ok);
+        if(lockTracker.isTargetLockOn()){
+            lockTracker.checkLock(ok.size());
+        }else{
+            lockTracker.setupLock(ok.size(), index);
+        }        
+        ArrayList<RotatedRect> selected = new FinalVisionTargetFilter().filter(ok, index);
```

Comment:
it looks like you will always use the result of CenterRectangleIndexFinder.getIndex().

I think you need a call to lockTracker.getSelectedIndex() somewhere dont you?

I think CenterRectangleFinder() should only be used when targetLock is not on?